v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java

```diff
@@ -442,6 +527,128 @@ private static List<String> parseResultSet(
     }
   }
 
+  /** Struct printer for converting a Spanner Struct to JSON. */
+  public static class StructJSONPrinter implements StructPrinter {
+    private StructValidator structValidator = null;
+
+    public StructJSONPrinter(StructValidator structValidator) {
+      this.structValidator = structValidator;
+    }
+
+    public StructJSONPrinter() {}
+
+    /**
+     * To get string in json format from Struct.
+     *
+     * @param struct Spanner Struct.
+     * @return Spanner Struct encoded as a JSON String.
+     */
+    @Override
+    public String print(Struct struct) {
+      if (structValidator != null) {
+        structValidator.validate(struct);
+      }
+
+      StringWriter stringWriter = new StringWriter();
+      try {
+        JsonWriter jsonWriter = new JsonWriter(stringWriter);
+        LinkedHashMap<String, BiFunction<Struct, String, String>> parsers = Maps.newLinkedHashMap();
+        parsers.putAll(mapColumnParsers(struct.getType().getStructFields()));
+        parseResultSet(jsonWriter, struct, parsers);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
nit adding some details to the exception is recommended 

e.g.,
throw new RuntimeException("Error parsing the Spanner struct to JSON", e);